### PR TITLE
fix flake8-simplify failures caused by recently added checks.

### PR DESCRIPTION
**Pull request by wxtim**
This is a small change with no associated Issue.
Recent changes to [flake8 Simplify](https://github.com/MartinThoma/flake8-simplify/commits/master) highlighted some over-complex logic in Cylc. This PR simplifies this logic.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.py` and
  `conda-environment.yml`.
- [x] Already covered by existing tests, i.e. flake8
- [x] No change log entry required - no functionality change
- [x] No documentation update required.

### CODE REVIEW

File: cylc/flow/task_job_mgr.py [hunk 1]
```diff
@@ -815,7 +815,7 @@ def _poll_task_job_callback(self, workflow, itask, cmd_ctx, line):
                 itask, INFO, FAIL_MESSAGE_PREFIX + jp_ctx.run_signal,
                 jp_ctx.time_run_exit,
                 flag)
-        elif jp_ctx.run_status == 1:
```
Comment (oliver-sanders) [inline cylc/flow/task_job_mgr.py:818]: This conversion is not quite identical as it causes the `jp_ctx.time_run` evaluation to happen earlier in the `if/elif` block.

In this case we can revert to the way it was before and add an inline ignore comment.

File: cylc/flow/platforms.py [hunk 2]
```diff
@@ -88,7 +88,7 @@ def get_platform(
             get_platform() or platform_from_job_info(), but to the
             user these look the same.
     """
-    if task_conf is None or isinstance(task_conf, str):
+    if task_conf is None or isinstance(task_conf, str):  # noqa: SIM 114
```
Comment (wxtim) [inline cylc/flow/platforms.py:91]: I'm not sure that the QA is right about the suggested simplification.

Comment (oliver-sanders) [reply cylc/flow/platforms.py:91]: It isn't, there is an issue open on flake8-simplify.

**Review by oliver-sanders — APPROVED** [review]

**Comment by oliver-sanders** [comment]
Happy to single review this, all tests pass.

File: cylc/flow/task_pool.py [hunk 3]
```diff
@@ -625,14 +625,19 @@ def release_runahead_task(self, itask: TaskProxy) -> None:
...
+            if (
+                (
+                    not parent_points
+                    or all(x < self.config.start_point for x in parent_points)
+                )
+                or
+                (
+                    itask.tdef.get_abs_triggers(next_point)
+                )
+            ):
```
Comment (hjoliver) [inline cylc/flow/task_pool.py:637]: (this one already fixed on the integer flow label branch ... looking back after rebase conflicts on that PR!)

**Pull request merged**